src/semidiscretization/semidiscretization.jl

```diff
@@ -82,16 +82,35 @@ end
 
 Wrap the semidiscretization `semi` as an ODE problem in the time interval `tspan`
 that can be passed to `solve` from the [SciML ecosystem](https://diffeq.sciml.ai/latest/).
+
+The optional keyword arguments `storage_type` and `real_type` configure the underlying computational
+datastructures. `storage_type` changes the fundamental array type being used, allowing the
+experimental use of `CuArray` or other GPU array types. `real_type` changes the computational data type being used.
 """
 function semidiscretize(semi::AbstractSemidiscretization, tspan;
-                        reset_threads = true)
+                        reset_threads = true,
+                        storage_type = nothing,
+                        real_type = nothing)
     # Optionally reset Polyester.jl threads. See
     # https://github.com/trixi-framework/Trixi.jl/issues/1583
     # https://github.com/JuliaSIMD/Polyester.jl/issues/30
     if reset_threads
         Polyester.reset_threads!()
     end
 
+    if !(storage_type === nothing && real_type === nothing)
+        if storage_type === nothing
+            storage_type = Array
+        end
+        if real_type === nothing
+            real_type = Float64
+        end
```

Comment:
The `semi` itself is constructed with two arguments, `RealT` and `uEltype`, that determine the type of internal caches, etc. By default, they will be `Float64`, but there should be some clear interaction between all these types.

For example, how would you see the interaction with AD like in https://github.com/trixi-framework/Trixi.jl/blob/36dab43ad09975a27fe27958ab74f3b102898784/src/semidiscretization/semidiscretization.jl#L268-L278